pages/[username]/posts/[postId].tsx

```diff
@@ -89,8 +89,11 @@ export const PostPage = ({}: Props) => {
   };
 
   useEffect(() => {
-    fetchData();
-  }, [router.query]);
+    const timer = setTimeout(() => {
+      fetchData();
+    }, 1000);
+    return () => clearTimeout(timer);
+  }, [username, postId]);
```

Comment:
We do not want to be using a timeout inside useEffect. A timeout does not scale according to the internet speed. Would you have another way to solve the issue?

Reply:
No worries this time I have implemented conditionals that will execute functions based on specific conditions.